src/transformers/models/t5/modeling_t5.py

```diff
@@ -595,6 +595,17 @@ def __init__(self, config, has_relative_attention_bias=False):
 
         self.layer.append(T5LayerFF(config))
 
+        self.model_parallel = False
+
+    def block_parallelize(self, device):
+        self.to(device)
+        self.model_parallel = True
+
+    def block_deparallelize(self):
+        self.to("cpu")
+        self.model_parallel = False
```

Comment:
Do we need `model_parallel` as an attribute on the module itself?

Reply:
yes, if we use the automatic input remapper, since there is no other way to do `if self.model_parallel` but we surely will eventually remove anything that is not being used.

Actually do we really need `block_deparallelize`? What would be the practical use?